core/src/Streamly/Internal/Data/Stream/Chunked.hs

```diff
@@ -735,8 +735,10 @@ parseBreakK (PRD.Parser pstep initial extract) stream = do
                     arr0 = A.fromListN n (Prelude.reverse src0)
                     arr1 = Array contents next end
                     str = K.cons arr0 (K.fromPure arr1)
-                return (b, str)
-            PR.Error err -> throwM $ ParseError err
+                return (Right b, str)
+            PR.Error err -> do
+                let arr = A.fromList (Prelude.reverse backBuf)
+                return (Left (ParseError err), K.fromPure arr)
```

Comment:
cannot return the backtracking buffer on error, need to return the forward array buf instead.